src/jinja2/lexer.py

```diff
@@ -509,7 +509,7 @@ def c(x):
             TOKEN_COMMENT_BEGIN: [
                 (
                     c(
-                        fr"(.*?)((?:\-{comment_end_re}\s*"
+                        fr"(.*?)(\+{comment_end_re}|(?:\-{comment_end_re}\s*"
```

Comment:
This regex is inconsistent with the additions in other places - I do not know the lexer though is this on purpose?

Reply:
Are you referring to how this one was not wrapped in a non-capturing group? 
I noticed the original `((?:\-{comment_end_re}` had an outermost capture group unlike the others so I kept it that way for the addition.

Reply:
Keeping the outer group is good, but I think they mean that previously `comment_end_re` was never captured, now it would sometimes be captured.

Reply:
Yes, I would have expected
```diff
- fr"(.*?)((?:\-{comment_end_re}\s*"
+ fr"(.*?)(?:\+{comment_end_re}|(?:\-{comment_end_re}\s*"
```
though on a second look at the parentheses It may even be more fitting to use
```diff
- fr"(.*?)((?:\-{comment_end_re}\s*"
+ fr"(.*?)((?:\+{comment_end_re}|\-{comment_end_re}\s*"
```

Reply:
Ohh I see. Thank you, will make the changes! From your second look @septatrix , this means the parentheses for the other regex additions can also be simplified right?

Reply:
I would assume so, yes - though you might have moved them too far back by one group (- again I did not run the test so this may also work ;D )

Reply:
I see the inconsistency here between this block and the others. Even before the change this one captured the suffix, while the others ignored it. The change made it ignored as well. If the tests are still passing, I'd stick with that for consistency.

Reply:
The tests are still passing. So stick with the changes and have them all ignore the the suffix?

Reply:
Yeah, that makes sense to me.